### PR TITLE
Added strong typing for the following API methods

### PR DESCRIPTION
**Pull request by schaefi**
* storage/subformat/vagrant_base.py
* storage/subformat/vagrant_libvirt.py
* storage/subformat/vagrant_virtualbox.py
* storage/subformat/vdi.py
* storage/subformat/vhd.py
* storage/subformat/vhdfixed.py
* storage/subformat/vhdx.py
* storage/subformat/vmdk.py

This references Issue #1644

### CODE REVIEW

File: kiwi/storage/subformat/vagrant_base.py [hunk 1]
```diff
@@ -222,7 +220,7 @@ def get_additional_vagrant_config_settings(self):
         :return: additional vagrant settings
         :rtype: str
         """
-        pass
+        return ''
 
     def _create_box_metadata(self):
         metadata = self.get_additional_metadata() or {}
```
Comment (jesusbv) [inline kiwi/storage/subformat/vagrant_base.py:226]: unnecessary `or {}` ? It returns an empty dict now

Comment (schaefi) [reply kiwi/storage/subformat/vagrant_base.py:226]: yes any implementation of the function in a child class returns a dict. The abstract interface class however returned None. So the type is broken. The same for the other interface class which is expected to return a string

Comment (schaefi) [reply kiwi/storage/subformat/vagrant_base.py:226]: see implementations of vagrant_libvirt.py and vagrant_virtualbox.py for details

Comment (schaefi) [reply kiwi/storage/subformat/vagrant_base.py:226]: pythons ```if not``` operator will continue to work in the same way as with the None type before. Thus there was no behavior change and the base class methods are not called in the context of vagrant anyway

Comment (schaefi) [reply kiwi/storage/subformat/vagrant_base.py:226]: That's why I thought the change is ok and imho the only way to make static type checks happy. An alternative would be ```->  Optional[...]``` but did not like it that much in this case. Thoughts ?

Comment (davidcassany) [reply kiwi/storage/subformat/vagrant_base.py:226]: I am fine with this current implementation. I also prefer avoiding `Optional[...]`

**Review by davidcassany — APPROVED** [review]

**Pull request merged**